frontend/lib/modules/lend_library/lend_library.dart

```diff
@@ -168,13 +168,9 @@ class _LendLibraryState extends State<LendLibrary> {
 
   void _init() async {
     if (!_skipCurrentLocation) {
-      var coordinates = await _location.getLocation();
-      if (coordinates.latitude != null) {
-        setState(() {
-          _filters['lat'] = coordinates.latitude!;
-          _filters['lng'] = coordinates.longitude!;
-        });
-      }
+      List<dynamic> _userLngLat = await Provider.of<CurrentUserState>(context, listen: false).getUserLocation();
```

Comment:
In this case, we want to get current location wherever they are (even if not at their default location). E.g. if they are 5 miles away or in a different location, we want to show close items, not only items close to their home location. So in this case we ALWAYS want to get current location. BUT to make things load fast, we can start with a default to home and then after we get current location, can compare, and if they are different enough, we can re-query with the new location.

Reply:
Gotcha. In that case we can render the page, but let the km away render till location info is ready. i.e. take location out of page init, right?